test/openAPI/features/support/oauth_token.js

```diff
@@ -0,0 +1,216 @@
+    code,
+    clientId,
+    clientAssertionType,
+    redirectUri,
+  ) => {
+    specOAuthToken.post(baseUrl).withJson({
+      grant_type: grantType,
+      client_assertion_type: clientAssertionType,
+      client_assertion: null,
+      client_id: clientId,
+      code: code,
+      redirect_uri: redirectUri,
+    })
+});
+
+Then(
+  /^The POST \/oauth\/token endpoint response should contain "([^"]*)" as errorType$/,
+  (errorCode) =>
+    chai
+      .expect(specOAuthToken._response.json.error)
+      .to.be.equal(errorCode)
+);
+
+// Scenario: The user is not able to receive the ID and access token because of the invalid client_assertion_type
+// Given, Then for this scenario are written in the aforementioned example
+When(
+  /^The User sends POST request with given "([^"]*)" as grantType, "([^"]*)" as code, "([^"]*)" as clientId, "([^"]*)" as clientAssertion, "([^"]*)" as redirectUri$/,
+  (
+    grantType,
+    code,
+    clientId,
+    clientAssertion,
+    redirectUri,
+  ) => {
+    specOAuthToken.post(baseUrl).withJson({
+      grant_type: grantType,
+      client_assertion_type: null,
+      client_assertion: clientAssertion,
+      client_id: clientId,
+      code: code,
+      redirect_uri: redirectUri,
+    })
+});
+
+
+// Scenario: The user is not able to receive the ID and access token because of the invalid redirect_uri
+// Given, Then for this scenario are written in the aforementioned example
+When(
+  /^The User sends POST request with given "([^"]*)" as grantType, "([^"]*)" as code, "([^"]*)" as clientId, "([^"]*)" as clientAssertionType, "([^"]*)" as clientAssertion$/,
+  (
+    grantType,
+    code,
+    clientId,
+    clientAssertion,
+    clientAssertionType,
+  ) => {
+    specOAuthToken.post(baseUrl).withJson({
+      grant_type: grantType,
+      client_assertion_type: clientAssertionType,
+      client_assertion: clientAssertion,
+      client_id: clientId,
+      code: code,
+      redirect_uri: null,
+    })
+});
+
+
+// Scenario: The user is not able to receive the ID and access token because of the invalid input
+// Given, Then for this scenario are written in the aforementioned example
+When(
+  /^The User sends POST request with given "([^"]*)" as grantType, "([^"]*)" as code, "([^"]*)" as clientId, "([^"]*)" as clientAssertionType, "([^"]*)" as clientAssertion, "([^"]*)" as redirectUri$/,
+  (
+    grantType,
+    code,
+    clientId,
+    clientAssertion,
+    clientAssertionType,
+    redirectUri
+      ) => {
+    specOAuthToken.post(baseUrl).withJson({
+      grant_type: grantType,
+      client_assertion_type: clientAssertionType,
+      client_assertion: clientAssertion,
+      client_id: clientId,
+      code: code,
+      redirect_uri: redirectUri,
+    })
+});
+
+
+// Scenario: The user is not able to receive the ID and access token because of the empty payload
+// Given, Then for this scenario are written in the aforementioned example
+When(
+  /^The User sends POST request with empty payload$/,
```

Comment:
I'd suggest sending the request without .withJson if you want to send the req without payload.
By sending json that has null values you still send a payload, and that mismatches with scenario name.